server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java

```diff
@@ -104,23 +101,16 @@ public void start(C configuration, EmbeddedCacheManager cacheManager) {
          throw new IllegalStateException("CacheIgnoreManager is a required component");
       }
 
-      executor = new ThreadPoolExecutor(
-            configuration.workerThreads(),
-            configuration.workerThreads(),
-            0L, TimeUnit.MILLISECONDS,
-            new LinkedBlockingQueue<>(),
-            new DefaultThreadFactory(getQualifiedName() + "-ServerHandler"),
-            new ThreadPoolExecutor.AbortPolicy() {
-               @Override
-               public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
-                  if (e.isShutdown())
-                     throw new IllegalLifecycleStateException("Server has been stopped");
-                  else
-                     super.rejectedExecution(r, e);
-               }
-            });
-
-      manageableThreadPoolExecutorService = new ManageableThreadPoolExecutorService(executor);
+      executor = bcr.getComponent(KnownComponentNames.BLOCKING_EXECUTOR, ExecutorService.class).running();
+
+      if (executor instanceof BlockingTaskAwareExecutorServiceImpl) {
```

Comment:
when is this `true`? 
I think we only create `LazyInitializingBlockingTaskAwareExecutorService`. It is already a `ManageableExecutorService` so it can be used directly without more wrapping.

Reply:
Hrmm, I had an issue in testing, maybe it isn't a problem anymore. Let me try again.

Reply:
I have removed the if, if CI passes then it should be good ;)